generic-linux/make-installer.sh

```diff
@@ -3,198 +3,76 @@
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstcoreelements.so      ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstcoreelements.so      ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstcoreindexers.so      ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstdecodebin.so         ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstdecodebin2.so        ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstencodebin.so         ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstffmpegcolorspace.so  ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstflump3dec.so         ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstgdp.so               ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstgio.so               ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstlibvisual.so         ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstnice010.so           ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstogg.so               ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstpango.so             ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstplaybin.so           ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstplaybin.so           ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstpulse.so             ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstsubparse.so          ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgsttcp.so               ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgsttheora.so            ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgsttypefindfunctions.so ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstvideorate.so         ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstvideoscale.so        ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstvideotestsrc.so      ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstvolume.so            ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstvorbis.so            ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstximagesink.so        ./build/lib
-    # cp /usr/lib/x86_64-linux-gnu/gstreamer-0.10/libgstxvimagesink.so       ./build/lib
+mkdir -p build/lib/luasql
+for lib in lfs rex_pcre luasql/sqlite3 zip lua-utf8
+do
+  for path in $(lua -e "print(package.cpath)" | tr ";" "\n")
+  do
+    changed_path=${path/\?/${lib}};
+    if [ -e "${changed_path}" ]; then
+      cp -rL "${changed_path}" build/lib/${lib}.so
+    fi
+  done
+done
+
+# copy in files ignored (?) by linuxdeployqt
+cp "$(ldd build/lib/rex_pcre.so | cut -d ' ' -f 3 | grep 'libpcre')" build/lib
```

Comment:
These files are ignored because it's not a direct compile-time dependency - they are loaded at runtime by those extra Lua libraries (which we load at runtime ourselves), so it has no way of knowing they're needed.